### ISSUE TITLE
v2 is broken on arm64 macOS runners

### ISSUE THREAD

**Issue by nazar-pc** [KyleMayes/install-llvm-action#65]
I tested both self-hosted and GitHub's runners and in all cases when trying to install LLVM action detects x64 instead of arm64:
```
Run KyleMayes/install-llvm-action@bb56a8457a77b95a5fa576c23731cbb162fd2056
  with:
    version: 17
  env:
    CARGO_INCREMENTAL: 0
    CARGO_TERM_COLOR: always
    RUSTFLAGS: -C strip=symbols -C opt-level=s --cfg aes_armv8
hello
Error: Unsupported version for platform (os=darwin, arch=x64, version=17)!
Checking known assets (os=darwin, arch=x64, version=17)...
    at $73b874f14e970995$export$da446a5ee2572b7 (/Users/runner/work/_actions/KyleMayes/install-llvm-action/bb56a8457a77b95a5fa576c23731cbb162fd2056/dist/main.js:7615:41)
Error: Unsupported version for platform (os=darwin, arch=x64, version=17)!
    at $73b874f14e970995$var$install (/Users/runner/work/_actions/KyleMayes/install-llvm-action/bb56a8457a77b95a5fa576c23731cbb162fd2056/dist/main.js:7633:60)
    at $73b874f14e970995$export$889ea624f2cb2c57 (/Users/runner/work/_actions/KyleMayes/install-llvm-action/bb56a8457a77b95a5fa576c23731cbb162fd2056/dist/main.js:7663:16)
    at $933db609b41224a7$var$main (/Users/runner/work/_actions/KyleMayes/install-llvm-action/bb56a8457a77b95a5fa576c23731cbb162fd2056/dist/main.js:7681:61)
    at Object.<anonymous> (/Users/runner/work/_actions/KyleMayes/install-llvm-action/bb56a8457a77b95a5fa576c23731cbb162fd2056/dist/main.js:7688:1)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
```

P.S. Hello :slightly_smiling_face:

**Comment by KyleMayes** [comment]
[The architecture is currently determined by using the `arch` input if provided, then using NodeJS's `process.arch`, finally falling back to `x64` if `process.arch` is not available.](https://github.com/KyleMayes/install-llvm-action/blob/master/index.ts#L65)

So I think this means that `process.arch` is an empty string instead of `"arm64"` on these runners and I'm not sure why.

I think I'll need to come up with an alternate fallback for architecture detection.

In the meantime I guess you'll need to explicitly supply `arm64` to the `arch` input.

**Comment by nazar-pc** [comment]
We have tested and confirmed that `console.log(require('node:process').arch);` prints `arm64` at least on self-hosted runner. I was suspecting that we have accidentally installed x86-64 Node.js on that machine, but then confirmed that the same is happening with official runners too, so it is not our self-hosted environment that is the reason (logs provided are from official runner).

**Comment by KyleMayes** [comment]
I didn't realize `core.getInput` from the GitHub Actions library returns an empty string when an option isn't provided 🤦 

So `const arch = (options.arch ?? process.arch) || "x64";` was always resolving to `"x64"` since `options.arch` was always an empty string and `??` only returns the second argument when the first is `null` or `undefined`.

In my defense, this is at least half JavaScript's fault, something like this probably wouldn't happen in Rust.

I should have a fix out soon.

**Comment by KyleMayes** [comment]
Well I have architecture detection working properly now, but the installed Clang binaries aren't runningon the ARM64 macOS runners:

https://github.com/KyleMayes/install-llvm-action/actions/runs/8515915176/job/23324181025?pr=66

```
dyld[4329]: weak-def symbol not found '__ZnwmSt19__type_descriptor_t'weak-def symbol not found '__ZnamSt19__type_descriptor_t'
/Users/runner/work/_temp/b5afcbc1-d1b9-4e85-b786-8533e2ed7487.sh: line 3:  4329 Abort trap: 6           $CC --version
```

**Comment by nazar-pc** [comment]
> In my defense, this is at least half JavaScript's fault, something like this probably wouldn't happen in Rust.

Yes, I'm so happy I mostly write Rust these days and don't have to deal with such errors.

> Well I have architecture detection working properly now, but the installed Clang binaries aren't running properly on the ARM64 macOS runners:

According to https://discourse.cmake.org/t/cmake-suddenly-stopped-working-macos-sonoma/10303 and https://discourse.llvm.org/t/apples-libc-now-provides-std-type-descriptor-t-functionality-not-found-in-upstream-libc/73881 linked from there it is again some macOS-specific crap.

Looks like environment variables need to be edited for it to work.

**Comment by KyleMayes** [comment]
Prefixing `DYLD_LIBRARY_PATH` with `/usr/lib` seemed to do the trick, it's now working in a standard ARM64 macOS runner.

I've released `v2.0.1` with this fix and pointed `v2` at that.
